lib/galaxy/tools/actions/__init__.py

```diff
@@ -297,37 +297,51 @@ def execute(self, tool, trans, incoming=None, return_job=False, set_output_hid=T
         # Build name for output datasets based on tool name and input names
         on_text = self._get_on_text(inp_data)
 
-        # format='input" previously would give you a random extension from
-        # the input extensions, now it should just give "input" as the output
-        # format.
-        input_ext = 'data' if tool.profile < 16.04 else "input"
-        input_dbkey = incoming.get("dbkey", "?")
-        for name, data in reversed(list(inp_data.items())):
-            if not data:
-                data = NoneDataset(datatypes_registry=app.datatypes_registry)
-                continue
-
-            # Convert LDDA to an HDA.
-            if isinstance(data, LibraryDatasetDatasetAssociation) and not completed_job:
-                data = data.to_history_dataset_association(None)
-                inp_data[name] = data
-
-            if tool.profile < 16.04:
-                input_ext = data.ext
-
-            if data.dbkey not in [None, '?']:
-                input_dbkey = data.dbkey
-
-            identifier = getattr(data, "element_identifier", None)
-            if identifier is not None:
-                incoming["%s|__identifier__" % name] = identifier
-
-        # Collect chromInfo dataset and add as parameters to incoming
-        (chrom_info, db_dataset) = execution_cache.get_chrom_info(tool.id, input_dbkey)
-
-        if db_dataset:
-            inp_data.update({"chromInfo": db_dataset})
-        incoming["chromInfo"] = chrom_info
+        if tool.profile <= 21.01:
+
+            # format='input" previously would give you a random extension from
+            # the input extensions, now it should just give "input" as the output
+            # format.
+            input_ext = 'data' if tool.profile < 16.04 else "input"
+            input_dbkey = incoming.get("dbkey", "?")
+            for name, data in reversed(list(inp_data.items())):
+                if not data:
+                    data = NoneDataset(datatypes_registry=app.datatypes_registry)
+                    continue
+
+                # Convert LDDA to an HDA.
```

Comment:
Maybe move the legacy behavior into a separate function, for clarity ?

Reply:
good idea